internal/sqlutil/sql.go

```diff
@@ -153,6 +158,22 @@ func RunLimitedVariablesQuery(ctx context.Context, query string, qp QueryProvide
 	return nil
 }
 
+// RunLimitedVariablesExec split up a query with more variables than the used database can handle in multiple queries.
+func RunLimitedVariablesExec(ctx context.Context, query string, qp ExecProvider, variables []interface{}, limit uint) error {
```

Comment:
This absolutely needs tests. For sliding sync I use a different API - [Chunkify](https://github.com/matrix-org/sliding-sync/blob/main/sqlutil/sql.go#L49) but it ends up doing the same thing.